packages/fxa-payments-server/src/components/CouponForm/index.test.tsx

```diff
@@ -19,16 +21,16 @@ describe('CouponForm', () => {
     expect(couponButton).toBeInTheDocument();
   });
 
-  it('shows an error message when invalid coupon code is used', () => {
+  it('shows an error message when invalid coupon code is used', async () => {
     const subject = () => {
-      return render(<CouponForm coupon={undefined} setCoupon={() => {}} />);
+      return render(<CouponForm planId={SELECTED_PLAN.plan_id} coupon={undefined} setCoupon={() => {}} />);
     };
 
     const { queryByTestId, getByTestId } = subject();
     fireEvent.change(getByTestId('coupon-input'), { target: { value: 'a' } });
     fireEvent.click(getByTestId('coupon-button'));
 
-    expect(queryByTestId('coupon-error')).toBeInTheDocument();
+    await waitForExpect(() => expect(queryByTestId('coupon-error')).toBeInTheDocument());
```

Comment:
Another way is to wrap the code that causing React rendering in `act()`.  E.g. `await act(async () => fireEvent.click(someButton));`.

Reply:
So I tried doing that, but for some reason it wasn't working. It was late at night though, so I might have done something else wrong. 😅

Reply:
Works > doesn't work. 😄